app/controllers/ops_controller.rb

```diff
@@ -91,6 +91,43 @@ def x_button
     generic_x_button(OPS_X_BUTTON_ALLOWED_ACTIONS)
   end
 
+  def load_tags_categories
+    cats = Classification.eager_load(:tag).visible.writeable.categories.delete_if { |c| c.entries.length == 0 } # Get the categories, sort by description
+    data = cats.map{ |c| {:id =>c.id.to_s, :name => c.name, :description => c.description, :tag_id => c.tag_id.to_s } }
+    render :json => data
+  end
+
+  def get_category_entries
+    x = Classification.find_by_id(params[:cat_id])
+    render :json => x.entries.map { |entry| { :value => entry[:id], :label => entry[:description], :name => entry.tag[:name] } }
+  end
+
+  def get_category_entries_multi
+    category_entries = {}
+    params[:ids].map { |category_id|
+      cat = Classification.find_by_id(category_id)
```

Comment:
I believe that this could be done in query
`Classification.where(:ids =>  params[:ids]).includes(:tags)....`

I see similar pattern in such queries maybe you will find how to unify it to one method and put it on model.